docs/deploy-custom-image.rst

```diff
@@ -4,175 +4,151 @@
+.. _Docker: https://docs.docker.com/
 
-.. warning:: If you want to use Scrapy in the custom Docker image you may want to include
-    `scrapinghub-entrypoint-scrapy`_ in your requirements.txt file.
-    It is a support layer that passes data from the job to Scrapinghub storage.
-    Otherwise you will need to send data to Scrapinghub storage using `HTTP API`__.
-
-.. _scrapinghub-entrypoint-scrapy: https://pypi.python.org/pypi/scrapinghub-entrypoint-scrapy
-__ https://doc.scrapinghub.com/scrapy-cloud.html#storage-scrapinghub-com
-
-2. Define your target image
----------------------------
-Now you need to define the Docker repository that will store the image built by this tool. To do this, open your project's ``scrapinghub.yml`` file and add an ``image`` option to it, like this:
-
-.. code-block:: yaml
+Deployment
+==========
 
-    projects:
-        default: 29629
-    image: yourusername/repository
+This section describes how to build and deploy a custom Docker image to Scrapy Cloud. For all the following steps
+it's assumed that commands are executed at the root directory of your project.
 
+1. Create Dockefile
+-------------------
 
-The settings above define that shub will push the image of your Docker container to https://hub.docker.com/r/yourusername/repository. You can also specify the complete URL for your repository if you are not using the default registry (which is https://hub.docker.com).
+The most important thing you need to be able to build and deploy Docker images is a `Dockerfile`_.
+Please follow the link if you are not familiar with the concept as it's crucial to understand it
+while using custom Docker images feature.
 
-.. tip:: Your project might not have a ``scrapinghub.yml`` file, because it has been introduced with recent versions of shub. Make sure to upgrade your shub package by running::
+If you want to migrate an existing Scrapy project - there's a tool that may help you, please read
+`this section <create-image-for-scrapy-project>`_. In all other cases you're responsible for writing your own
+Dockerfile. The resulting Dockerfile should produce a Docker image that follows the
+:ref:`Custom images contract <custom-images-contract>` - follow the link to find an example Dockefile.
 
-            $ pip install shub --upgrade
+.. _Dockerfile: https://docs.docker.com/engine/reference/builder/
 
-    And then create ``scrapinghub.yml`` by running::
+2. Deploy to Scrapy Cloud
+-------------------------
 
-            $ shub deploy
+Once you have the Dockerfile run the :ref:`shub deploy <basic-usage>` command to build the Docker image.
+If there's no :ref:`scrapinghub.yml <configuration>` configuration file at the project root
+shub will start a wizard that will help to configure the project and will save the configuration file.
+If you already have :ref:`scrapinghub.yml <configuration>` at the project root please ensure that
+:ref:`image deploy is configured <configuration-options>` for the target project.
```

Comment:
I'm a bit reluctant to advertize `shub image upload` because I hope that eventually we will remove it and keep only `shub deploy`, but I'm okay to add this for now.